bindings/pydeck-carto/examples/notebooks/carto_layer.ipynb

```diff
@@ -0,0 +1,172 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f90a430c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "register_carto_layer()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9b703385",
+   "metadata": {},
+   "source": [
+    "## CARTO DW\n",
+    "\n",
+    "Every CARTO account has a **CARTO Data Warehouse** to work with. In this example, a table with world airports from the CARTO DW is rendered."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dfebc544",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "layer = pdk.Layer(\n",
+    "    \"CartoLayer\",\n",
+    "    data=\"carto-demo-data.demo_tables.world_airports\",\n",
+    "    type_=MapType.TABLE,\n",
+    "    connection=CartoConnection.CARTO_DW,\n",
+    "    credentials=get_layer_credentials(carto_auth),\n",
+    "    get_fill_color=[238, 77, 90],\n",
+    "    point_radius_min_pixels=2.5,\n",
+    "    pickable=True,\n",
+    ")\n",
+    "\n",
+    "view_state = pdk.ViewState(latitude=0, longitude=0, zoom=1)\n",
+    "\n",
+    "tooltip = {\"html\": \"<b>Name:</b> {name}\", \"style\": {\"color\": \"white\"}}\n",
+    "\n",
+    "pdk.Deck(\n",
+    "    layer, map_style=pdk.map_styles.ROAD, initial_view_state=view_state, tooltip=tooltip\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f89456a7",
+   "metadata": {},
+   "source": [
+    "## Data Warehouses\n",
+    "\n",
+    "CARTO allows rendering cloud data from **BigQuery, Snowflake, Redshift, Postgres and Databricks** using custom connections.\n",
+    "\n",
+    "This example performs a spatial intersection to display the airports in Spain. Note that it uses a BigQuery connection called \"bigquery\"."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "172e318c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "layer = pdk.Layer(\n",
+    "    \"CartoLayer\",\n",
+    "    data=\"\"\"\n",
+    "        SELECT a.geom, a.name\n",
+    "        FROM `carto-demo-data.demo_tables.world_airports` AS a,\n",
+    "             `carto-do-public-data.natural_earth.geography_glo_admin0countries_410` AS g\n",
+    "        WHERE g.ADMIN = 'Spain' AND\n",
+    "              ST_INTERSECTS(a.geom, g.geom)\n",
+    "    \"\"\",\n",
+    "    type_=MapType.QUERY,\n",
+    "    connection=pdk.types.String(\"bigquery\"),\n",
+    "    credentials=get_layer_credentials(carto_auth),\n",
+    "    get_fill_color=[238, 77, 90],\n",
+    "    point_radius_min_pixels=2.5,\n",
+    "    pickable=True,\n",
+    ")\n",
+    "\n",
+    "view_state = pdk.ViewState(latitude=36, longitude=-7.44, zoom=4)\n",
+    "\n",
+    "tooltip = {\"html\": \"<b>Name:</b> {name}\", \"style\": {\"color\": \"white\"}}\n",
+    "\n",
+    "pdk.Deck(\n",
+    "    layer, map_style=pdk.map_styles.ROAD, initial_view_state=view_state, tooltip=tooltip\n",
+    ")"
+   ]
+  },
+  {
```

Comment:
The file format is quite verbose and hard to work with, what with all the escaped characters and the inline image. It might be worth considering in the future to split this file into separate python script files & assets and assemble them using a script. It would make code review much simpler and we could lint the code

Reply:
Great, thanks for the tips :)